static/js/containers/VideoEmbedPage.js

```diff
@@ -27,28 +28,71 @@ class VideoEmbedPage extends React.Component<*, void> {
   }
 
   render() {
-    const { video, videoUi } = this.props
-
+    const { video } = this.props
     return (
       <DocumentTitle title={`OVS | ${video.title} | Video Embed`}>
-        <div className="embedded-video">
-          <VideoPlayer
-            video={video}
-            cornerFunc={this.updateCorner}
-            selectedCorner={videoUi.corner}
-            embed={true}
-          />
-        </div>
+        {this.renderBody()}
       </DocumentTitle>
     )
   }
+
+  renderBody () {
+    const { video, videoUi } = this.props
+    const videoStatus = this.getVideoStatus(video)
+    if (videoStatus === 'PROCESSING') {
+      return this.renderProcessingMessage()
+    }
+    else if (videoStatus === 'ERROR') {
+      return this.renderErrorMessage()
+    }
```

Comment:
why not just check `videoIsProcessing(video)` and `videoHasError(video)` right here, instead of having the separate `getVideoStatus` function?

Reply:
In the tests there's a `video` object created, can't you just mutate that to make the different test cases?

Reply:
Hm, I suppose there's a trade-off there. To me, writing the proxy function just to enable easier testing is undesirable, because we're trading off code readability for ease of testing.  But if you feel strongly this is the right way to go about it I'm fine with it.